python/kserve/kserve/storage/storage.py

```diff
@@ -68,47 +68,46 @@ def download(uri: str, out_dir: str = None) -> str:
         if uri.startswith(_PVC_PREFIX) and not os.path.exists(uri):
             raise Exception(f"Cannot locate source uri {uri} for PVC")
 
-        is_local = False
-        if uri.startswith(_LOCAL_PREFIX) or os.path.exists(uri):
-            is_local = True
-
-        if out_dir is None:
-            if is_local:
+        is_local = uri.startswith(_LOCAL_PREFIX) or os.path.exists(uri)
+        if is_local:
+            if out_dir is None:
                 # noop if out_dir is not set and the path is local
-                return Storage._download_local(uri)
-            out_dir = tempfile.mkdtemp()
-        elif not os.path.exists(out_dir):
-            os.mkdir(out_dir)
-
-        if uri.startswith(_GCS_PREFIX):
-            Storage._download_gcs(uri, out_dir)
-        elif uri.startswith(_S3_PREFIX):
-            Storage._download_s3(uri, out_dir)
-        elif uri.startswith(_HDFS_PREFIX) or uri.startswith(_WEBHDFS_PREFIX):
-            Storage._download_hdfs(uri, out_dir)
-        elif re.search(_AZURE_BLOB_RE, uri):
-            Storage._download_azure_blob(uri, out_dir)
-        elif re.search(_AZURE_FILE_RE, uri):
-            Storage._download_azure_file_share(uri, out_dir)
-        elif is_local:
-            return Storage._download_local(uri, out_dir)
-        elif re.search(_URI_RE, uri):
-            return Storage._download_from_uri(uri, out_dir)
-        elif uri.startswith(MODEL_MOUNT_DIRS):
-            # Don't need to download models if this InferenceService is running in the multi-model
-            # serving mode. The model agent will download models.
-            return out_dir
+                model_dir = Storage._download_local(uri)
+            else:
+                model_dir = Storage._download_local(uri, out_dir)
```

Comment:
Is there any risk that `out_dir` would not exist and this would fail?

The previous code seems to make sure that `out_dir` exists by creating it, if needed. But, now, for the local branch, it seems to assume that it would exist.

Reply:
Ahh! Missed the case. Fixed